index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
@izaakschroeder I spent some time wiring it up with the path logic (https://github.com/pillarjs/path-to-regexp/pull/50/files#diff-168726dbe96b3ce427e7fedce31bb0bcR98). Let me know what you think. It'll give you an idea of the various cases too.